lib/sqlalchemy/sql/elements.py

```diff
@@ -794,16 +821,50 @@ def reverse_operate(
         ) -> ColumnElement[Any]:
             ...
 
+        @overload
         def op(
             self,
             opstring: str,
             precedence: int = 0,
             is_comparison: bool = False,
-            return_type: Optional[_TypeEngineArgument[_OPT]] = None,
+            *,
+            return_type: _TypeEngineArgument[_OPT],
             python_impl: Optional[Callable[..., Any]] = None,
         ) -> Callable[[Any], BinaryExpression[_OPT]]:
             ...
 
```

Comment:
**mike bayer** (zzzeek) wrote:

OK just this part, what is this getting us?   I dont see any tests added for the op() change here so I dont know what this covers.  can we add something?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4794

Reply:
I think I answered this here here https://github.com/sqlalchemy/sqlalchemy/pull/10108#discussion_r1266297680
With the current implementation, if return_type is None, the overload that matches is the following:

```python
@overload
def op(
            self,
            opstring: str,
            precedence: int = 0,
            is_comparison: bool = False,
            return_type: Optional[_TypeEngineArgument[_OPT]] = None,
            python_impl: Optional[Callable[..., Any]] = None,
        ) -> Callable[[Any], BinaryExpression[_OPT]]:
  ...
```

However the `_OPT` used in the return type `Callable[[Any], BinaryExpression[_OPT]]` cannot be inferred.
mypy will fallback to _OPT=Any, but pyright will report that `_OPT` is unknown.

Reply:
If I remove the overload that has `*` in it, does a test fail ?  taht's all we need here